extensions/resource-deployment/src/ui/eulaWizard/eulaSettingsPage.ts

```diff
@@ -0,0 +1,576 @@
+			value: localize('eulaSettingsPage.AgreementCheckboxLabel', "Accept terms of use"),
+			CSSStyles: {
+				'font-weight': '300',
+				'font-size': '14px'
+			}
+		}).component();
+		const checkbox = this._view.modelBuilder.checkBox().withProperties<azdata.CheckBoxProperties>({
+			ariaLabel: this.getAgreementDisplayText(agreementInfo),
+			required: true
+		}).component();
+		checkbox.checked = false;
+		checkbox.onChanged(() => {
+			this._agreementCheckboxChecked = !!checkbox.checked;
+		});
+		const text = this._view.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: agreementInfo.template,
+			links: agreementInfo.links,
+			requiredIndicator: true
+		}).component();
+		return createFlexContainer(this._view, [title, createFlexContainer(this._view, [checkbox, text])], false);
+	}
+
+	private getAgreementDisplayText(agreementInfo: AgreementInfo): string {
+		// the agreement template will have {index} as placeholder for hyperlinks
+		// this method will get the display text after replacing the placeholders
+		let text = agreementInfo.template;
+		for (let i: number = 0; i < agreementInfo.links.length; i++) {
+			text = text.replace(`{${i}}`, agreementInfo.links[i].text);
+		}
+		return text;
+	}
+
+
+	private createOptionsDropdown(): azdata.FlexContainer {
+
+		const container = this._view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
+
+		const labels: string[] = [];
+
+		if (this._resourceType.options) {
+			this._resourceType.options.forEach((option, i) => {
+				const optionLabel = this._view.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+					value: option.displayName
+				}).component();
+				optionLabel.width = '150px';
+
+				labels.push(option.displayName);
+
+				let selectedValue = option.values[0];
+				if (this._presets) {
+					selectedValue = this._presets.dropdownValues![i];
+				}
+
+				const optionSelectBox = this._view.modelBuilder.dropDown().withProperties<azdata.DropDownProperties>({
+					values: option.values,
+					value: selectedValue,
+					width: '300px',
+					ariaLabel: option.displayName
+				}).component();
+
+				this._optionsDropdown.push(optionSelectBox);
+
+				optionSelectBox.onValueChanged(async () => {
+
+					let presetFields = new EulaInformation();
+					presetFields.dropdownValues = [];
+
+					this._optionsDropdown.forEach(element => {
+						presetFields?.dropdownValues?.push(element.value as azdata.CategoryValue);
+					});
+
+					this.wizard.changeProvider(this.getCurrentProvider(), presetFields);
+				});
+
+				this._optionDropDownMap.set(option.name, optionSelectBox);
+
+
+
+				const row = this._view.modelBuilder.flexContainer().withItems([optionLabel, optionSelectBox], { flex: '0 0 auto', CSSStyles: { 'margin-right': '20px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
+				container.addItem(row);
+			});
+		}
+
+		let containerLabelTest = 'Select ';
```

Comment:
How to localize this string

Reply:
What do you mean? And what is this even for?

Reply:
I was referring to this text. 
![image](https://user-images.githubusercontent.com/6816294/94531272-2dad2480-01f1-11eb-8ad7-e9c8ceb58914.png)

Reply:
So what's the question then? You localize the strings like any other.

Reply:
Aasim, look at localizedConstants.ts file in resource-deployment. You define a new constant using the examples in that file and then use the newly defined constant as the value of containerLabelTest. However, you are concatenating version and target labels. if you want to do that then you will need to create a function that returns a localized string. See examples in localizedConstants.ts in the azdata or the arc extension.